test/common/event/dispatcher_impl_test.cc

```diff
@@ -51,5 +53,77 @@ TEST(DispatcherImplTest, DeferredDelete) {
   dispatcher.clearDeferredDeleteList();
 }
 
+class DispatcherImplTest : public ::testing::Test {
+protected:
+  DispatcherImplTest() {
+    {
+      std::lock_guard<std::mutex> lock(mu_);
+      work_id_ = 0;
+    }
+    dispatcher_ = std::make_unique<DispatcherImpl>();
+    dispatcher_thread_ = std::make_unique<Thread::Thread>([this]() {
+      // Must create a keepalive timer to keep the dispatcher from exiting.
+      std::chrono::milliseconds time_interval(500);
+      keepalive_timer_ = dispatcher_->createTimer(
+          [this, time_interval]() { keepalive_timer_->enableTimer(time_interval); });
+      keepalive_timer_->enableTimer(time_interval);
+
+      dispatcher_->run(Dispatcher::RunType::Block);
+    });
+  }
+
+  ~DispatcherImplTest() override {
+    dispatcher_->exit();
+    dispatcher_thread_->join();
+  }
+
+  std::unique_ptr<Thread::Thread> dispatcher_thread_;
+  DispatcherPtr dispatcher_;
+  std::mutex mu_;
+  int work_id_;
+  TimerPtr keepalive_timer_;
+};
+
+TEST_F(DispatcherImplTest, Post) {
+  dispatcher_->post([this]() {
+    std::lock_guard<std::mutex> lock(mu_);
+    work_id_ = 1;
+  });
+
+  bool reached_value = false;
+  while (!reached_value) {
```

Comment:
nit: This is basically going to spin until the post happens. Would it be better to use a condition variable for these types of tests?